pkg/catalog/generator.go

```diff
@@ -101,13 +101,23 @@ func handleSourceURL(importPath string) string {
 					file := path.Join(paths[3:]...)
 
 					urlPath.Path = path.Join(paths[:3]...)
-					return path.Join(BuildGithubBlobURL(urlPath.String()), file)
+					urlPath, err = url.Parse(BuildGithubBlobURL(urlPath.String()))
```

Comment:
nit: Could possibly pull out `url.Parse(BuildGithubBlobURL(remoteURL, filePath))` into a function for better readability

Reply:
added cleaner solution